### PR TITLE
LPD-41846 Allow user to localize boolean fields

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-41846](https://liferay.atlassian.net/browse/LPD-41846)

[LPD-41846]: https://liferay.atlassian.net/browse/LPD-41846?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/BooleanObjectFieldBusinessType.java [hunk 1]
```diff
@@ -57,7 +58,7 @@ public PropertyDefinition.PropertyType getPropertyType() {
 
 	@Override
 	public boolean isLocalizable() {
-		return true;
+		return FeatureFlagManagerUtil.isEnabled("LPD-32050");
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/BooleanObjectFieldBusinessType.java:61]: @pedro-oliveira446 We should consider the companyId to check the feature flag. I see that we don't have this info in this level, so remove this check from here and place it just where it's being used.

**Review by gabrielwas — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/BooleanObjectFieldBusinessType.java (hunk 1 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/BooleanObjectFieldBusinessType.java]: Just done!

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 2]
```diff
@@ -915,13 +923,10 @@ private void _addObjectFieldColumn(
 	private void _addOrUpdateObjectFieldSettings(
 			ObjectDefinition objectDefinition, ObjectField newObjectField,
 			ObjectField oldObjectField,
+			ObjectFieldBusinessType objectFieldBusinessType,
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:927]: I think we need to sort these parameters. What do you think? Which pattern did you follow regarding sorting?

**Comment by marcelabc** [comment]
I review the PR and I just have this question :+1:

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java (hunk 2 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:926]: Just done!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 3]
```diff
@@ -913,15 +922,12 @@ private void _addObjectFieldColumn(
 	}
 
 	private void _addOrUpdateObjectFieldSettings(
-			ObjectDefinition objectDefinition, ObjectField newObjectField,
-			ObjectField oldObjectField,
-			List<ObjectFieldSetting> objectFieldSettings)
+			ObjectField newObjectField,
+			ObjectFieldBusinessType objectFieldBusinessType,
+			List<ObjectFieldSetting> objectFieldSettings,
+			ObjectDefinition objectDefinition, ObjectField oldObjectField)
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:928]: Sort

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java (hunk 3 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java]: Done!

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
Hey @pedro-oliveira446 
Even though the stable has failed (a known issue), you can check the results using the Testray link.
You can manually forward it to Brian if there are no unique failures.
Thank you!

**Comment by pedro-oliveira446** [comment]
ci:report:84645579

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by pedro-oliveira446** [comment]
ci:report:85498480

**Comment by pedro-oliveira446** [comment]
ci:report:85800864

**Comment by pedro-oliveira446** [comment]
Resent https://github.com/brianchandotcom/liferay-portal/pull/156743

**Pull request closed without merging**